uid/snowflake.go

```diff
@@ -68,6 +68,7 @@ func (u *ID) UnmarshalText(b []byte) error {
 	return nil
 }
 
-func (u *ID) MarshalText() ([]byte, error) {
+// TODO: test cases
+func (u ID) MarshalText() ([]byte, error) {
```

Comment:
I noticed this while testing. It's the same problem as #1625.

From [effective Go](https://go.dev/doc/effective_go#pointers_vs_values):
> The rule about pointers vs. values for receivers is that value methods can be invoked on pointers and values, but pointer methods can only be invoked on pointers.

By using a value receiver, we can marshal api types directly without them needing to be pointers. We don't use pointers to `uid.ID`, but I think what ends up happening is that `json.Marshal` calls `reflect.Value.CanAddr`. When that returns false it is not able to use the method with a pointer receiver.

I can write a similar set of tests for this in another PR.